code/modules/mob/living/simple_animal/guardian/guardian.dm

```diff
@@ -529,6 +530,9 @@ GLOBAL_LIST_EMPTY(parasites) //all currently existing/living guardians
 	if(guardians.len && !allowmultiple)
 		to_chat(user, "<span class='holoparasite'>You already have a [mob_name]!</span>")
 		return
+	if(HAS_TRAIT_FROM(user, TRAIT_NODEATH, "memento_mori"))									//checks if the user has an active mori link, if so, give fluff and cancel.
```

Comment:
there are other sources of TRAIT_NODEATH, though, including being a high-functioning zombie

Reply:
also, you can activate a holoparasite injector, then bind memento mori to yourself while it's polling ghosts for your holoparasite to get around the checks you've put in place here.

Reply:
the real solution here, IMO, is to make holoparasites consume you either when you die or when your health dips below -100.

Reply:
exactly what i've been saying, this problem will pop up again whenever another source of NODEATH comes around if it's solved with a specific check instead of a proper generic solution

Reply:
I guess I could try to do that